### DISCUSSION TITLE
Why are my authorization provider variables getting overridden?

### DISCUSSION THREAD

**Discussion by scherven** [supabase/supabase#30557] [Questions] [answered]
I am attempting to set up google authorization in a supabase preview branch. I copied over the client ID and client key variables that work perfectly fine in my production branch. However, now whenever I commit to the branch, they get reset to placeholders somehow, which then break the authentification flow since they have illegal characters:
<img width="860" alt="image" src="https://github.com/user-attachments/assets/68cd6d33-4369-4972-89da-00ba04b5bf30">
How can I prevent this? What is even causing this?

**Comment by silentworks** [comment]
This might be a bug in the latest CLI as it now handles copying the configs from your local to your remote project. I don't have the branching feature on any of my project so I'm unable to test this. I would suggest opening an issue on the https://github.com/supabase/cli repo about this.

**Comment by sweatybridge** [comment]
Thanks for reporting this issue. We will fix it so that you can continue setting up external auth provider on preview branch via dashboard.

Meanwhile, if you have any `[auth.external]` blocks in config.toml, please comment them out or delete them so it won't be picked up by branching.

**Reply by scherven** [reply]
Is there any way to set this up so it works automatically? I don't want to have to manually add back in my client secret to every preview branch.

**Reply by sweatybridge** [reply, marked as answer]
We have just added support for encrypting secrets in config.toml using dotenvx. If you declare them following [our example](https://github.com/supabase/supabase/tree/master/examples/slack-clone/nextjs-slack-clone-dotenvx#core-concept), the auth secrets will be synced to your preview branches securely.
